### PR TITLE
Add Cloud.ca machine driver

### PR DESCRIPTION
**Pull request by erikwilson**
This change adds the Cloud.ca machine driver to the list of drivers
available by default for installation with Rancher.

For /issues/15444

### CODE REVIEW

**Comment by pdube** [comment]
Thanks @erikwilson! When I added the machine driver locally, I had to add a whitelist domain for objects-east.cloud.ca. Do we need to add something here for the whitelisting?

**Comment by erikwilson** [comment]
There is a whitelist in this PR for `*.cloud.ca`, which should allow for `api.cloud.ca` and `objects-east.cloud.ca`, we can define those two explicitly instead of using `*` if you are sure those will be the only two endpoints used. Unfortunately it looks like there may be a minor bug where we do not consider whitelists when checking for changes in a machine driver, so if you already had a definition for `cloudca` the whitelist would not be updated. If the machine driver is removed and Rancher server restarted the built-in whitelist and driver values should be re-populated.

**Comment by pdube** [comment]
Great thanks!

**Comment by pdube** [comment]
I updated the driver with the fixes. (Here)[https://github.com/rancher/rancher/issues/15444#issuecomment-454068404] are the new values. Thanks for opening the PR

File: app/machinedriver_data.go [hunk 1]
```diff
@@ -1,60 +1,91 @@
...
 import (
+	"reflect"
+
 	"github.com/rancher/types/apis/management.cattle.io/v3"
 	"github.com/rancher/types/config"
 	"github.com/sirupsen/logrus"
 	"k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
+type machineDriverCompare struct {
```
[the remark this answers is not in the document]
Comment (erikwilson) [reply app/machinedriver_data.go]: The `NodeDriverSpec` contains additional fields, such as `DisplayName`, `Description`, and `ExternalID`, which might trigger unintended updates, or if modified to zero out / copy those fields any future changes to the Spec might contain additional fields which trigger unintended updates.

**Comment by nathan-jenan-rancher** [comment]
LGTM

**Comment by cjellick** [comment]
egad @erikwilson ! conflict. you have to resolve. take a look at what changed and figure out if it affect what you did

**Comment by erikwilson** [comment]
Resolved, added annotations to new/old comparisons

**Comment by cjellick** [comment]
LGTM

**Comment by cjellick** [comment]
@sangeethah we want the majority of the QA for these third-party drivers to be performed by the third-party. As this was our first one in awhile, I had @erikwilson go a bit further to ensure the framework and driver generally worked.
But I think your team should do very little validation beyond asserting that the driver is available and can be turned on.
cc @deniseschannon

**Pull request merged**